tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -4657,6 +4690,61 @@ def _assert_is_fit(self, message_suffix: str = None):
                 error_message = f"{error_message} `.{message_suffix}`."
             raise AssertionError(error_message)
 
+    @Deprecated(min_version_to_warn="0.8", min_version_to_error="1.1", version_to_remove="1.1", new="persist")
```

Comment:
What about the risk that erroring in v1.1 will break semantic versioning for AutoGluon?

For example, if a user upgrades to v1.0 and for some reason misses the DeprecationWarning (e.g., code is invoked outside of `if __name__ == "__main__":`, so deprecations are silenced [by default](https://peps.python.org/pep-0565/)), they could get upset when the code breaks after the v1.1 upgrade.

Reply:
Do you have an alternative proposal? Does Pandas/NumPy do it differently?

Reply:
Pandas seems to always introduce deprecations in minor releases, and use major releases for breaking changes (and no new deprecations) https://pandas.pydata.org/docs/development/policies.html#version-policy. It seems that this wouldn't apply to us since our next release is a major version.

The way I see it, we can either 1/ break the API now or 2/ keep the old method names around until v2.0 to follow semantic versioning.

Reply:
Discussed offline and decided on removing in v1.2